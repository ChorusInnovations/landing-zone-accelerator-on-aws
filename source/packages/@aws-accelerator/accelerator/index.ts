/**
 *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/accelerator';
export * from './lib/config-repository';
export * from './lib/pipeline';
export * from './lib/stacks/accounts-stack';
export * from './lib/stacks/default-stack';
export * from './lib/stacks/dependencies-stack';
export * from './lib/stacks/network-1-stack';
export * from './lib/stacks/network-2-stack';
export * from './lib/stacks/network-3-stack';
export * from './lib/stacks/operations-stack';
export * from './lib/stacks/organizations-stack';
export * from './lib/stacks/pipeline-stack';
export * from './lib/stacks/security-audit-stack';
export * from './lib/stacks/security-stack';
export * from './lib/stacks/validate-stack';
export * from './lib/toolkit';
