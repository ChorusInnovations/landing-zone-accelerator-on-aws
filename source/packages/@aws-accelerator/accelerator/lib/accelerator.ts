/**
 *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */
import { AccountsConfig, GlobalConfig, OrganizationConfig, SecurityConfig } from '@aws-accelerator/config';
import { throttlingBackOff } from '@aws-accelerator/utils';
import { AssumeProfilePlugin } from '@aws-cdk-extensions/cdk-plugin-assume-role';
import { DescribeOrganizationCommand, OrganizationsClient, paginateListAccounts } from '@aws-sdk/client-organizations';
import { RequireApproval } from 'aws-cdk/lib/diff';
import { PluginHost } from 'aws-cdk/lib/plugin';
import { AcceleratorToolkit } from './toolkit';

export enum AcceleratorStage {
  PIPELINE = 'pipeline',
  /**
   * Validate Stage - Verify the configuration files and environment
   */
  VALIDATE = 'validate',
  ORGANIZATIONS = 'organizations',
  LOGGING = 'logging',
  /**
   * Accounts Stage - Handle all Organization and Accounts actions
   */
  ACCOUNTS = 'accounts',
  DEPENDENCIES = 'dependencies',
  SECURITY = 'security',
  OPERATIONS = 'operations',
  NETWORKING = 'networking',
  'SECURITY-AUDIT' = 'security-audit',
}

/**
 *
 */
export interface AcceleratorProps {
  command: string;
  configDirPath: string;
  parallel: boolean;
  stage: string;
  account: string;
  region: string;
  partition: string;
  requireApproval: RequireApproval;
}

/**
 * Wrapper around the CdkToolkit. The Accelerator defines this wrapper to add
 * the following functionality:
 *
 * - x
 * - y
 * - z
 */
export abstract class Accelerator {
  static isSupportedStage(stage: AcceleratorStage): boolean {
    return Object.values(AcceleratorStage).includes(stage);
  }

  /**
   *
   * @returns
   */
  static async run(props: AcceleratorProps): Promise<void> {
    //
    // When an account and region is specified, execute as single stack
    //
    if (props.account || props.region) {
      if (props.account && props.region === undefined) {
        throw new Error(`Account set to ${props.account}, but region is undefined`);
      }
      if (props.region && props.account === undefined) {
        throw new Error(`Region set to ${props.region}, but region is undefined`);
      }
      return await AcceleratorToolkit.execute({
        command: props.command,
        accountId: props.account,
        region: props.region,
        stage: props.stage,
        configDirPath: props.configDirPath,
        requireApproval: props.requireApproval,
      });
    }

    //
    // Read in all Accelerator Configuration files here, then pass the objects
    // to the stacks that need them. Exceptions are thrown if any of the
    // configuration files are malformed.
    //
    const globalConfig = GlobalConfig.load(props.configDirPath);
    const organizationsConfig = OrganizationConfig.load(props.configDirPath);
    const accountsConfig = AccountsConfig.load(props.configDirPath);
    const securityConfig = SecurityConfig.load(props.configDirPath);

    //
    // Load Plugins
    //
    const assumeRolePlugin = new AssumeProfilePlugin({
      // TODO: Read this from arg
      assumeRoleName: organizationsConfig['organizations-access-role'],
      assumeRoleDuration: 3600,
    });
    assumeRolePlugin.init(PluginHost.instance);

    //
    // NOTE: We do some early environment validation here before we kick off the
    //       CodePipeline that has a built in validation-stack
    //

    //
    // Verify AWS Organizations has been enabled
    //
    const organizationsClient = new OrganizationsClient({});
    await throttlingBackOff(() =>
      organizationsClient.send(new DescribeOrganizationCommand({})).catch(error => {
        if (error.name === 'AWSOrganizationsNotInUseException') {
          throw new Error(error.message);
        }
        throw new Error(error);
      }),
    );

    //
    // Create a dictionary of all AWS Account IDs.
    //
    // TODO: Add functionality to read in a map from file for disconnected regions
    //
    const accountIds: { [name: string]: string } = {};
    for await (const page of paginateListAccounts({ client: organizationsClient }, {})) {
      for (const account of page.Accounts ?? []) {
        if (account.Email && account.Id) {
          accountIds[account.Email] = account.Id;
        }
      }
    }

    //
    // Execute Bootstrap stacks for all identified accounts
    //
    if (props.command == 'bootstrap') {
      const trustedAccountId = accountIds[accountsConfig['mandatory-accounts'].management.email];
      for (const region of globalConfig['enabled-regions']) {
        for (const account of Object.values(accountsConfig['mandatory-accounts'])) {
          const accountId = accountIds[account.email];
          await AcceleratorToolkit.execute({
            command: props.command,
            accountId,
            region,
            partition: props.partition,
            trustedAccountId,
            requireApproval: props.requireApproval,
          });
        }
        for (const account of Object.values(accountsConfig['workload-accounts'])) {
          const accountId = accountIds[account.email];
          await AcceleratorToolkit.execute({
            command: props.command,
            accountId,
            region,
            partition: props.partition,
            trustedAccountId,
            requireApproval: props.requireApproval,
          });
        }
      }
      return;
    }

    // TODO: Need to decide the mandatory accounts for an accelerator --
    // Control Tower: To start a well-planned OU structure in your landing zone, AWS Control Tower
    // sets up a Security OU for you. This OU contains three shared accounts: the management
    // (primary) account, the log archive account, and the security audit account (also referred to
    // as the audit account).

    switch (props.stage) {
      case AcceleratorStage.VALIDATE:
      case AcceleratorStage.ACCOUNTS:
        break;
      case AcceleratorStage.ORGANIZATIONS:
        const managementAccount = accountsConfig['mandatory-accounts'].management;
        for (const region of globalConfig['enabled-regions']) {
          await AcceleratorToolkit.execute({
            command: props.command,
            accountId: accountIds[managementAccount.email],
            region: region,
            stage: props.stage,
            configDirPath: props.configDirPath,
            requireApproval: props.requireApproval,
          });
        }
        break;

      //
      // Apply these stacks to all account / regions. The contents of these stacks are dynamically
      // built from the inputted configuration files during stack construction
      //
      case AcceleratorStage.LOGGING:
      case AcceleratorStage.DEPENDENCIES:
      case AcceleratorStage.SECURITY:
        for (const region of globalConfig['enabled-regions'].filter(
          item => securityConfig.getExcludeRegions()?.indexOf(item) === -1,
        )) {
          for (const account of Object.values(accountsConfig['mandatory-accounts'])) {
            console.log(`${region} to be implemented on for ${account.email} account`);
            const accountId = accountIds[account.email];
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId,
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
          for (const account of Object.values(accountsConfig['workload-accounts'])) {
            const accountId = accountIds[account.email];
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId,
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
        }
        break;
      case AcceleratorStage.OPERATIONS:
      case AcceleratorStage.NETWORKING:
        for (const region of globalConfig['enabled-regions']) {
          for (const account of Object.values(accountsConfig['mandatory-accounts'])) {
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId: accountIds[account.email],
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
          for (const account of Object.values(accountsConfig['workload-accounts'])) {
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId: accountIds[account.email],
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
        }
        break;
      case AcceleratorStage['SECURITY-AUDIT']:
        const auditAccountName = securityConfig.getDelegatedAccountName();
        if (accountsConfig.accountExists(auditAccountName)) {
          for (const region of globalConfig['enabled-regions'].filter(
            item => securityConfig.getExcludeRegions()?.indexOf(item) === -1,
          )) {
            const auditAccountEmail = accountsConfig.getEmail(auditAccountName);
            const accountId = accountIds[auditAccountEmail];
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId,
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
        } else {
          throw new Error(`Security audit delegated account ${auditAccountName} not found.`);
        }
        break;
      default:
        throw new Error(`Unknown stage: ${props.stage}`);
    }
  }
}
