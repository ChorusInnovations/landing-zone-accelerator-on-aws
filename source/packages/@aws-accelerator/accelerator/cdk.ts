/**
 *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as config from '@aws-accelerator/config';
import * as fs from 'fs';
import mri from 'mri';
import process from 'process';
import { Accelerator } from './lib/accelerator';
import { AcceleratorStage } from './lib/accelerator-stage';
import { AcceleratorToolkit, AcceleratorToolkitCommand } from './lib/toolkit';

process.on('unhandledRejection', (reason, _) => {
  console.error(reason);
  // eslint-disable-next-line no-process-exit
  process.exit(1);
});

const usage = `Usage: cdk.ts <command> --stage STAGE --config-dir CONFIG_DIRECTORY [--account ACCOUNT] [--region REGION] [--parallel]`;

const args = mri(process.argv.slice(2), {
  boolean: ['parallel'],
  string: ['account'],
  alias: {
    c: 'config-dir',
    s: 'stage',
    a: 'account',
    r: 'region',
    p: 'partition',
  },
  default: {
    parallel: false,
  },
});

const commands = args['_'];
const parallel = args['parallel'];
const configDirPath = args['config-dir'];
const stage = args['stage'];
const account = args['account'];
const region = args['region'];
const partition = args['partition'];
const requireApproval = args['require-approval'];

//
// Validate args: must specify a command
//
if (commands.length === 0) {
  console.log('<command> not set');
  throw new Error(usage);
}

//
// Validate args: verify command against our sub-list
//
if (!AcceleratorToolkit.isSupportedCommand(commands[0])) {
  throw new Error(`Invalid command: ${commands[0]}`);
}

//
// Validate args: verify stage if not bootstrap
//
if (!Accelerator.isSupportedStage(stage) && commands[0] !== String(AcceleratorToolkitCommand.BOOTSTRAP)) {
  throw new Error(`Invalid stage: ${stage}`);
}

//
// Validate args: verify config directory
//
if (stage !== AcceleratorStage.PIPELINE) {
  if (config === undefined || !fs.existsSync(configDirPath)) {
    console.log(`Invalid --config-dir ${configDirPath}`);
    throw new Error(usage);
  }
}

//
// Execute the Accelerator engine
//
Accelerator.run({
  command: commands[0],
  configDirPath,
  stage,
  parallel,
  account,
  region,
  partition,
  requireApproval,
}).catch(function (err) {
  console.log(err.message);
  process.exit(1);
});
