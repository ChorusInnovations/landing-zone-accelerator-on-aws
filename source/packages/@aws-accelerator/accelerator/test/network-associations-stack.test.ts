/**
 *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import {
  AccountsConfig,
  GlobalConfig,
  IamConfig,
  NetworkConfig,
  OrganizationConfig,
  SecurityConfig,
} from '@aws-accelerator/config';
import * as cdk from 'aws-cdk-lib';
import * as path from 'path';
import { AcceleratorStackNames } from '../lib/accelerator';
import { AcceleratorStage } from '../lib/accelerator-stage';
import { AcceleratorStackProps } from '../lib/stacks/accelerator-stack';
import { NetworkAssociationsStack } from '../lib/stacks/network-associations-stack';

const testNamePrefix = 'Construct(NetworkAssociationsStack): ';

/**
 * NetworkAssociationsStack
 */
const app = new cdk.App({
  context: { 'config-dir': path.join(__dirname, 'configs/all-enabled') },
});
const configDirPath = app.node.tryGetContext('config-dir');

const props: AcceleratorStackProps = {
  configDirPath,
  accountsConfig: AccountsConfig.load(configDirPath),
  globalConfig: GlobalConfig.load(configDirPath),
  iamConfig: IamConfig.load(configDirPath),
  networkConfig: NetworkConfig.load(configDirPath),
  organizationConfig: OrganizationConfig.load(configDirPath),
  securityConfig: SecurityConfig.load(configDirPath),
  partition: 'aws',
};

/**
 * Build all related stacks
 */
const stacks = new Map<string, NetworkAssociationsStack>();

for (const region of props.globalConfig.enabledRegions) {
  for (const account of [...props.accountsConfig.mandatoryAccounts, ...props.accountsConfig.workloadAccounts]) {
    const accountId = props.accountsConfig.getAccountId(account.name);

    stacks.set(
      `${account.name}-${region}`,
      new NetworkAssociationsStack(
        app,
        `${AcceleratorStackNames[AcceleratorStage.NETWORK_VPC]}-${accountId}-${region}`,
        {
          env: {
            account: accountId,
            region,
          },
          ...props,
        },
      ),
    );
  }
}

/**
 * NetworkAssociationsStack construct test
 */
describe('NetworkAssociationsStack', () => {
  /**
   * Number of Lambda function resource test
   */
  test(`${testNamePrefix} Lambda function resource count test`, () => {
    cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).resourceCountIs('AWS::Lambda::Function', 4);
  });

  /**
   * Number of Lambda function IAM role resource test
   */
  test(`${testNamePrefix} Lambda function IAM role resource count test`, () => {
    cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).resourceCountIs('AWS::IAM::Role', 4);
  });

  // /**
  //  * Number of SSM parameter resource test
  //  */
  // test(`${testNamePrefix} SSM parameter resource count test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).resourceCountIs('AWS::SSM::Parameter', 2);
  // });

  // /**
  //  * Number of TransitGatewayRouteTablePropagation resource test
  //  */
  // test(`${testNamePrefix} TransitGatewayRouteTablePropagation resource count test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).resourceCountIs(
  //     'AWS::EC2::TransitGatewayRouteTablePropagation',
  //     3,
  //   );
  // });

  // /**
  //  * Number of TransitGatewayRouteTableAssociation resource test
  //  */
  // test(`${testNamePrefix} TransitGatewayRouteTablePropagation resource count test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).resourceCountIs(
  //     'AWS::EC2::TransitGatewayRouteTableAssociation',
  //     1,
  //   );
  // });

  // /**
  //  * Number of GetTransitGatewayAttachment custom resource test
  //  */
  // test(`${testNamePrefix} GetTransitGatewayAttachment custom resource count test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).resourceCountIs(
  //     'Custom::GetTransitGatewayAttachment',
  //     1,
  //   );
  // });

  // /**
  //  * Cloudformation parameters resource configuration test
  //  */
  // test(`${testNamePrefix} Cloudformation parameters resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Parameters: {
  //       SsmParameterValueacceleratornetworktransitGatewaysMainidC96584B6F00A464EAD1953AFF4B05118Parameter: {
  //         Default: '/accelerator/network/transitGateways/Main/id',
  //         Type: 'AWS::SSM::Parameter::Value<String>',
  //       },
  //       SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablescoreidC96584B6F00A464EAD1953AFF4B05118Parameter:
  //         {
  //           Default: '/accelerator/network/transitGateways/Main/routeTables/core/id',
  //           Type: 'AWS::SSM::Parameter::Value<String>',
  //         },
  //       SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablessegregatedidC96584B6F00A464EAD1953AFF4B05118Parameter:
  //         {
  //           Default: '/accelerator/network/transitGateways/Main/routeTables/segregated/id',
  //           Type: 'AWS::SSM::Parameter::Value<String>',
  //         },
  //       SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablessharedidC96584B6F00A464EAD1953AFF4B05118Parameter:
  //         {
  //           Default: '/accelerator/network/transitGateways/Main/routeTables/shared/id',
  //           Type: 'AWS::SSM::Parameter::Value<String>',
  //         },
  //       SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablesstandaloneidC96584B6F00A464EAD1953AFF4B05118Parameter:
  //         {
  //           Default: '/accelerator/network/transitGateways/Main/routeTables/standalone/id',
  //           Type: 'AWS::SSM::Parameter::Value<String>',
  //         },
  //     },
  //   });
  // });

  // /**
  //  * Lambda function CustomGetTransitGatewayAttachmentCustomResourceProviderHandler resource configuration test
  //  */
  // test(`${testNamePrefix} Lambda function CustomGetTransitGatewayAttachmentCustomResourceProviderHandler resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       CustomGetTransitGatewayAttachmentCustomResourceProviderHandler7E079354: {
  //         Type: 'AWS::Lambda::Function',
  //         DependsOn: ['CustomGetTransitGatewayAttachmentCustomResourceProviderRoleA6A22C3D'],
  //         Properties: {
  //           Code: {
  //             S3Bucket: 'cdk-hnb659fds-assets-111111111111-us-east-1',
  //           },
  //           Handler: '__entrypoint__.handler',
  //           MemorySize: 128,
  //           Role: {
  //             'Fn::GetAtt': ['CustomGetTransitGatewayAttachmentCustomResourceProviderRoleA6A22C3D', 'Arn'],
  //           },
  //           Runtime: 'nodejs14.x',
  //           Timeout: 900,
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * Lambda function IAM role CustomGetTransitGatewayAttachmentCustomResourceProviderRole resource configuration test
  //  */
  // test(`${testNamePrefix} Lambda function IAM role CustomGetTransitGatewayAttachmentCustomResourceProviderRole resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       CustomGetTransitGatewayAttachmentCustomResourceProviderRoleA6A22C3D: {
  //         Type: 'AWS::IAM::Role',
  //         Properties: {
  //           AssumeRolePolicyDocument: {
  //             Statement: [
  //               {
  //                 Action: 'sts:AssumeRole',
  //                 Effect: 'Allow',
  //                 Principal: {
  //                   Service: 'lambda.amazonaws.com',
  //                 },
  //               },
  //             ],
  //             Version: '2012-10-17',
  //           },
  //           ManagedPolicyArns: [
  //             {
  //               'Fn::Sub': 'arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole',
  //             },
  //           ],
  //           Policies: [
  //             {
  //               PolicyDocument: {
  //                 Statement: [
  //                   {
  //                     Action: ['sts:AssumeRole'],
  //                     Effect: 'Allow',
  //                     Resource: '*',
  //                   },
  //                 ],
  //                 Version: '2012-10-17',
  //               },
  //               PolicyName: 'Inline',
  //             },
  //           ],
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * SSM parameter SsmParamStackId resource configuration test
  //  */
  // test(`${testNamePrefix} SSM parameter SsmParamStackId resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SsmParamStackId521A78D3: {
  //         Type: 'AWS::SSM::Parameter',
  //         Properties: {
  //           Name: '/accelerator/AWSAccelerator-NetworkAssociationsStack-111111111111-us-east-1/stack-id',
  //           Type: 'String',
  //           Value: {
  //             Ref: 'AWS::StackId',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * TransitGatewayRouteTablePropagation TestCorePropagation resource configuration test
  //  */
  // test(`${testNamePrefix} TransitGatewayRouteTablePropagation TestCorePropagation resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       TestCorePropagationB97A6DBD: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTablePropagation',
  //         Properties: {
  //           TransitGatewayAttachmentId: {
  //             Ref: 'TestVpcTransitGatewayAttachmentA903FB56',
  //           },
  //           TransitGatewayRouteTableId: {
  //             Ref: 'SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablescoreidC96584B6F00A464EAD1953AFF4B05118Parameter',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * TransitGatewayRouteTablePropagation TestSegregatedPropagation resource configuration test
  //  */
  // test(`${testNamePrefix} TransitGatewayRouteTablePropagation TestSegregatedPropagation resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       TestSegregatedPropagationCA3F8CD1: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTablePropagation',
  //         Properties: {
  //           TransitGatewayAttachmentId: {
  //             Ref: 'TestVpcTransitGatewayAttachmentA903FB56',
  //           },
  //           TransitGatewayRouteTableId: {
  //             Ref: 'SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablessegregatedidC96584B6F00A464EAD1953AFF4B05118Parameter',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * TransitGatewayRouteTableAssociation TestSharedAssociation resource configuration test
  //  */
  // test(`${testNamePrefix} TransitGatewayRouteTableAssociation TestSharedAssociation resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       TestSharedAssociation1890469B: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTableAssociation',
  //         Properties: {
  //           TransitGatewayAttachmentId: {
  //             Ref: 'TestVpcTransitGatewayAttachmentA903FB56',
  //           },
  //           TransitGatewayRouteTableId: {
  //             Ref: 'SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablessharedidC96584B6F00A464EAD1953AFF4B05118Parameter',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * TransitGatewayRouteTablePropagation TestSharedPropagation resource configuration test
  //  */
  // test(`${testNamePrefix} TransitGatewayRouteTablePropagation TestSharedPropagation resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       TestSharedPropagation66A144ED: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTablePropagation',
  //         Properties: {
  //           TransitGatewayAttachmentId: {
  //             Ref: 'TestVpcTransitGatewayAttachmentA903FB56',
  //           },
  //           TransitGatewayRouteTableId: {
  //             Ref: 'SsmParameterValueacceleratornetworktransitGatewaysMainrouteTablessharedidC96584B6F00A464EAD1953AFF4B05118Parameter',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * GetTransitGatewayAttachment TestVpcTransitGatewayAttachment resource configuration test
  //  */
  // test(`${testNamePrefix} GetTransitGatewayAttachment TestVpcTransitGatewayAttachment resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).templateMatches({
  //     Resources: {
  //       TestVpcTransitGatewayAttachmentA903FB56: {
  //         Type: 'Custom::GetTransitGatewayAttachment',
  //         UpdateReplacePolicy: 'Delete',
  //         DeletionPolicy: 'Delete',
  //         Properties: {
  //           ServiceToken: {
  //             'Fn::GetAtt': ['CustomGetTransitGatewayAttachmentCustomResourceProviderHandler7E079354', 'Arn'],
  //           },
  //           name: 'Test',
  //           roleArn: {
  //             'Fn::Join': [
  //               '',
  //               [
  //                 'arn:',
  //                 {
  //                   Ref: 'AWS::Partition',
  //                 },
  //                 ':iam::222222222222:role/AWSAccelerator-DescribeTgwAttachRole-us-east-1',
  //               ],
  //             ],
  //           },
  //           transitGatewayId: {
  //             Ref: 'SsmParameterValueacceleratornetworktransitGatewaysMainidC96584B6F00A464EAD1953AFF4B05118Parameter',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });
});
