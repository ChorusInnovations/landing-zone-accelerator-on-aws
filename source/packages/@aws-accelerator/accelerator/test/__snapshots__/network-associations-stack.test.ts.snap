// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkAssociationsStack Construct(NetworkAssociationsStack):  Snapshot Test 1`] = `
Object {
  "Resources": Object {
    "AcceleratorKeyLookup0C18DA36": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "CustomSsmGetParameterValueCustomResourceProviderLogGroup780D220D",
      ],
      "Properties": Object {
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomSsmGetParameterValueCustomResourceProviderHandlerAAD0E7EE",
            "Arn",
          ],
        },
        "assumeRoleArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::222222222222:role/AWSAccelerator-CrossAccount-SsmParameter-Role",
            ],
          ],
        },
        "invokingAccountID": "111111111111",
        "invokingRegion": "us-east-1",
        "parameterAccountID": "222222222222",
        "parameterName": "/accelerator/kms/key-arn",
        "parameterRegion": "us-east-1",
        "uuid": "REPLACED-UUID",
      },
      "Type": "Custom::SsmGetParameterValue",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomSsmGetParameterValueCustomResourceProviderHandlerAAD0E7EE": Object {
      "DependsOn": Array [
        "CustomSsmGetParameterValueCustomResourceProviderRoleB3AFDDB2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": "cdk-hnb659fds-assets-111111111111-us-east-1",
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomSsmGetParameterValueCustomResourceProviderRoleB3AFDDB2",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomSsmGetParameterValueCustomResourceProviderLogGroup780D220D": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "CustomSsmGetParameterValueCustomResourceProviderHandlerAAD0E7EE",
              },
            ],
          ],
        },
        "RetentionInDays": 3653,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomSsmGetParameterValueCustomResourceProviderRoleB3AFDDB2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "ssm:GetParameters",
                    "ssm:GetParameter",
                    "ssm:DescribeParameters",
                  ],
                  "Effect": "Allow",
                  "Resource": Array [
                    "*",
                  ],
                  "Sid": "SsmGetParameterActions",
                },
                Object {
                  "Action": Array [
                    "sts:AssumeRole",
                  ],
                  "Effect": "Allow",
                  "Resource": Array [
                    Object {
                      "Fn::Join": Array [
                        "",
                        Array [
                          "arn:",
                          Object {
                            "Ref": "AWS::Partition",
                          },
                          ":iam::*:role/AWSAccelerator*",
                        ],
                      ],
                    },
                  ],
                  "Sid": "StsAssumeRoleActions",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SsmParamAcceleratorVersionFF83282D": Object {
      "Properties": Object {
        "Name": "/accelerator/AWSAccelerator-NetworkAssociationsStack-111111111111-us-east-1/version",
        "Type": "String",
        "Value": "1.1.0",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "SsmParamStackId521A78D3": Object {
      "Properties": Object {
        "Name": "/accelerator/AWSAccelerator-NetworkAssociationsStack-111111111111-us-east-1/stack-id",
        "Type": "String",
        "Value": Object {
          "Ref": "AWS::StackId",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
  },
}
`;
